extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js

```diff
@@ -364,21 +376,31 @@ export class AmpLightboxGallery extends AMP.BaseElement {
         toggle(dev().assertElement(this.descriptionBox_), false);
       });
     } else {
-      const measureOverflow = state => {
-        state.isOverflown = this.descriptionBox_./*OK*/clientHeight
-          !== this.descriptionBox_./*OK*/scrollHeight;
-        state.overflowOn = this.descriptionBox_.classList.contains('overflow');
+      const measureOverflowState = state => {
+        // The height of the description without overflow is set to 4 rem.
+        // The height of the overflow mask is set to 1 rem. We allow 3 lines
+        // for the description and consider it to have overflow if more than 3
+        // lines of text.
+        state.hasOverflow = this.descriptionBox_./*OK*/scrollHeight
+            - this.descriptionBox_./*OK*/clientHeight
+            >= this.descriptionOverflowMask_./*OK*/clientHeight;
       };
 
-      const mutateOverflowIfApplicable = state => {
-        if (state.isOverflown || state.overflowOn) {
-          toggle(dev().assertElement(this.descriptionOverflowMask_), true);
-        } else if (!state.overflowOn) {
-          toggle(dev().assertElement(this.descriptionOverflowMask_), false);
+      const mutateOverflowState = state => {
+        // We toggle visibility instead of display because we rely on the height
+        // of this element to measure 1 rem.
+        if (state.hasOverflow) {
```

Comment:
nit: This could be simplified like this to eliminate the branches for readability

```js
st.setStyles(dev().assertElement(this.descriptionOverflowMask_), {
  visibility: state.hasOverflow ? 'visible' : 'hidden',
});
```